### ISSUE TITLE
Native mobile app testing

### ISSUE THREAD

**Issue by slavingia** [anti-work/shortest#227]
### What
- **Native mobile app testing support** in Shortest for iOS and Android.  
- Core workflows to install, launch, and interact with real/emulated apps using Shortest’s plain-English test syntax.  
- Additional setup docs or scripts to simplify environment configuration for mobile testing.

### Why
- Users need the same straightforward, AI-assisted test coverage for native mobile apps that Shortest offers for web.  
- Reduces QA overhead by allowing devs to reuse the existing Shortest approach (plain English tests) across all platforms.  
- Unlocks the **\$2,000 bounty** for the first contributor to implement iOS + Android support, payable in cash, Anti-Work (Gumroad, Inc.) equity, or a combo.

**Comment by gladyshcodes** [comment]
Maybe worth creating new `shortest-mobile1 repo for it?

**Comment by slavingia** [comment]
Why? I think one repo is much less cognitive load and seems reasonable to have it be able to test all 3 platforms using computer use.

**Comment by gladyshcodes** [comment]
> Why? I think one repo is much less cognitive load and seems reasonable to have it be able to test all 3 platforms using computer use

Ok

One more question: do you think that is a good idea to spin up expo react native repo? we need env to run mobile Shortest tests

**Comment by slavingia** [comment]
Yeah not sure the approach but something like that for sure will be useful.

**Comment by rmarescu** [comment]
@gladyshcodes has a draft PR https://github.com/anti-work/shortest/pull/260. Probalby best to collaborate on that one, to reduce duplicate effort.

**Comment by gladyshcodes** [comment]
Following up on our conversation with @rmarescu, we’ve decided to split #260 into several atomic PRs to reduce cognitive load, lower the risk of bugs, and ensure clearer, more focused code reviews:

Ensuring SRP:
- [ ] Refactor repository into a modular monolith architecture
- [ ] Create WebDriver and WebBrowser, deprecate BrowserManager and BrowserTool

Scaling for mobile:
- [ ] Mobile repo setup
- [ ] Add support for Android; AndroidDriver and AndroidBrowser
  - [ ] Appium environment setup
  - [ ] Image resizing caps for Anthropic models compatibility
- [ ] Add support for IOS; IOSDriver and IOSBrowser

Testing:
- [ ] In addition to exploratory testing of the above prs, write Vitest tests for core functionalities across all three platforms supported (after integrating Vitest into the codebase). Our goal is to achieve strong test coverage over time.

Final:
- [ ] Finalize implementation by addressing technical debt, refining modularity, and polishing for production readiness

Miscellaneous:
- [ ] Add JS shim
- [ ] Make `__shortest__` global variable with proper typing
- [ ] Convert anthropic tools to dynamic functions

cc @slavingia

**Comment by rmarescu** [comment]
I think we can discuss technical details next week, and scope for best approach. For now, we should pause development on this until we have a better foundation in place.

**Comment by gladyshcodes** [comment]
@rmarescu, I remember you mentioned that `Browser` isn't the best name. I'd love to hear your thoughts on these alternatives:

AndroidInterface
AndroidController
AndroidUIDriver (may be confusing as we already have `Driver` suffix)
AndroidSession (Manager)
AndroidNavigator
AndroidContext
AndroidAgent
AndroidPage (just as playwright)

**Comment by crabest** [comment]
instead of another repo, wouldn't be better to have a seperated folder for mobile with its core functions just my draft https://github.com/anti-work/shortest/pull/230/files ?

![Image](https://github.com/user-attachments/assets/9be92183-8fa1-4f2d-a632-bd1aaaaa359a)

**Comment by gladyshcodes** [comment]
> I think we can discuss technical details next week, and scope for best approach. For now, we should pause development on this until we have a better foundation in place.

Yep!

**Comment by gladyshcodes** [comment]
> instead of another repo, wouldn't be better to have a seperated folder for mobile with its core functions just my draft https://github.com/anti-work/shortest/pull/230/files ?

We need our own playground to test things. Regarding folder structure, we will most probably have a separate `driver` package ([see](https://github.com/anti-work/shortest/pull/260/files#diff-369f11350b6ee2ca4626bad60261cae325490243fc6940d940ecbb9816600df9R2))
